tests/func/test_analytics.py

```diff
@@ -1,65 +1,15 @@
-import os
-
 import mock
-import requests
+import json
 
-from dvc.analytics import Analytics
 from dvc.main import main
-from tests.basic_env import TestDir
-from tests.basic_env import TestDvc
-from tests.basic_env import TestGit
-
-
-def _clean_getenv(key, default=None):
-    """
-    Remove env vars that affect dvc behavior in tests
-    """
-    if key in ["DVC_TEST", "CI"]:
-        return None
-    return os.environ.get(key, default)
-
-
-class TestAnalytics(TestDir):
-    def test(self):
-        a = Analytics()
-        a.collect()
-        self.assertTrue(isinstance(a.info, dict))
-        self.assertNotEqual(a.info, {})
-        self.assertTrue(a.PARAM_USER_ID in a.info.keys())
-        self.assertTrue(a.PARAM_SYSTEM_INFO in a.info.keys())
-        self.assertNotEqual(a.info[a.PARAM_SYSTEM_INFO], {})
-
-    @mock.patch.object(os, "getenv", new=_clean_getenv)
-    @mock.patch("requests.post")
-    def test_send(self, mockpost):
-        ret = main(["daemon", "analytics", Analytics().dump(), "-v"])
-        self.assertEqual(ret, 0)
-
-        self.assertTrue(mockpost.called)
-
-    @mock.patch.object(os, "getenv", new=_clean_getenv)
-    @mock.patch.object(
-        requests, "post", side_effect=requests.exceptions.RequestException()
-    )
-    def test_send_failed(self, mockpost):
-        ret = main(["daemon", "analytics", Analytics().dump(), "-v"])
-        self.assertEqual(ret, 0)
-
-        self.assertTrue(mockpost.called)
-
-
-class TestAnalyticsGit(TestAnalytics, TestGit):
-    pass
-
+from dvc.utils.compat import str
 
-class TestAnalyticsDvc(TestAnalytics, TestDvc):
-    @mock.patch("requests.post")
-    def test_send_disabled(self, mockpost):
-        ret = main(["config", "core.analytics", "false"])
-        self.assertEqual(ret, 0)
 
-        with mock.patch.object(os, "getenv", new=_clean_getenv):
-            ret = main(["daemon", "analytics", Analytics().dump(), "-v"])
-        self.assertEqual(ret, 0)
+@mock.patch("dvc.analytics.send")
+def test_daemon_analytics(mock_send, tmp_path):
+    report = {"name": "dummy report"}
+    fname = tmp_path / "report"
+    fname.write_text(str(json.dumps(report)))
 
-        self.assertFalse(mockpost.called)
+    assert 0 == main(["daemon", "analytics", str(fname)])
```

Comment:
```suggestion
    assert 0 == main(["daemon", "analytics", fspath(fname)])
```
`str()` is for presentation, `fspath()` is for business) - to make it a str filename.